File: BOJ_1284/BOJ_1284.cs
using System;
using System.Linq;
using System.Text;
namespace asharien{
    static class PROGRAM{
        public static void Main(){
            StringBuilder SB = new StringBuilder();
            while(true){
                var N = Console.ReadLine();
                if(N == "0") break;
                int ANS = 0;
                foreach(char i in N){
                    if(i == '1') ANS +=2;
                    else if(i == '0') ANS += 4;
                    else ANS +=3;
                }
                SB.AppendLine((ANS+N.Length+1).ToString("G"));
            }
            Console.Write(SB);
        }
    }
}

File: BOJ_1284/BOJ_1284.py
import sys
GAP = {'1':2, '0':4}
while(True):
    ANS = 0
    N = str(sys.stdin.readline().rstrip())
    if(N == '0'):break
    for i in N:
        if i in GAP: ANS += GAP[i]
        else: ANS += 3
    print(ANS+len(N)+1)
